lib/internal/fs/utils.js

```diff
@@ -173,6 +173,29 @@ function copyObject(source) {
   return target;
 }
 
+function join(path, name) {
+  if ((typeof path === 'string' || path instanceof Buffer) &&
+    name === undefined) {
+    return path;
+  }
+
+  if (typeof path === 'string' && name instanceof Buffer) {
+    const pathBuffer = Buffer.from(pathModule.join(path, pathModule.sep));
+    return Buffer.concat([pathBuffer, name]);
+  }
+
+  if (typeof path === 'string' && typeof name === 'string') {
+    return pathModule.join(path, name);
+  }
+
+  if (path instanceof Buffer && name instanceof Buffer) {
+    return Buffer.concat([path, Buffer.from(pathModule.sep), name]);
```

Comment:
You could store the `Buffer.from(pathModule.sep)` buffer in a separate variable, so that we can avoid creating a new Buffer here each time

Reply:
done